datafusion/proto/src/physical_plan/from_proto.rs

```diff
@@ -340,24 +339,10 @@ pub fn parse_physical_expr(
             convert_required!(e.arrow_type)?,
         )),
         ExprType::ScalarFunction(e) => {
-            let scalar_function =
-                protobuf::ScalarFunction::try_from(e.fun).map_err(|_| {
-                    proto_error(
-                        format!("Received an unknown scalar function: {}", e.fun,),
-                    )
-                })?;
-
-            let args = parse_physical_exprs(&e.args, registry, input_schema, codec)?;
-
-            // TODO Do not create new the ExecutionProps
-            let execution_props = ExecutionProps::new();
-
-            functions::create_builtin_physical_expr(
-                &(&scalar_function).into(),
-                &args,
-                input_schema,
-                &execution_props,
-            )?
+            return Err(proto_error(format!(
```

Comment:
Since the only value left for ScalarFunction in the datafusion.proto is 'unknown' this implementation seemed reasonable.

Reply:
Maybe we could remove the ScalarFunction part of proto too 

In a follow on PR we could probably remove 

https://github.com/apache/datafusion/blob/70db5eab8996af4816958f798f6ee887dffb69ed/datafusion/proto/proto/datafusion.proto#L541-L686

And the references to `ScalarFunctionNode`